File: app/src/main/java/ua/hospes/rtm/core/di/module/ActivitiesModule.java
package ua.hospes.rtm.core.di.module;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import ua.hospes.rtm.ui.MainActivity;
import ua.hospes.rtm.ui.race.detail.RaceItemDetailActivity;

/**
 * @author Andrew Khloponin
 */
@Module
public abstract class ActivitiesModule {
    @ContributesAndroidInjector(modules = {FragmentsModule.class})
    abstract MainActivity contributeMainActivityInjector();

    @ContributesAndroidInjector(modules = {DomainModule.class})
    abstract RaceItemDetailActivity contributeRaceItemDetailActivityInjector();
}

File: app/src/main/java/ua/hospes/rtm/core/di/module/ActivitiesModule.kt
package ua.hospes.rtm.core.di.module

import dagger.Module
import dagger.android.ContributesAndroidInjector
import ua.hospes.rtm.ui.MainActivity
import ua.hospes.rtm.ui.race.detail.RaceItemDetailActivity

@Suppress("unused")
@Module
abstract class ActivitiesModule {
    @ContributesAndroidInjector(modules = [FragmentsModule::class])
    internal abstract fun bindMainActivity(): MainActivity

    @ContributesAndroidInjector(modules = [DomainModule::class])
    internal abstract fun bindRaceItemDetailActivity(): RaceItemDetailActivity
}
